python/tests/test_combinatorics.py

```diff
@@ -23,8 +23,13 @@
 """
 Test cases for combinatorial algorithms.
 """
+import io
 import itertools
 import unittest
+from collections import Counter
```

Comment:
I prefer to just import the module and then use ``collections.Counter`` in the code. This makes it obvious where the ``Counter`` class comes from. Otherwise, you see a ``Counter`` class in the code and you're not quite sure if it's the standard library's one, or a home grown version.